dmoj/cptbox/ptbox.h

```diff
@@ -29,6 +29,12 @@
 #include <seccomp.h>
 #endif
 
+#ifdef PTBOX_USE_LANDLOCK
```

Comment:
Thinking more about it, we likely do need runtime detection of Landlock capabilities.

With Landlock, we need to pass in `ALLOW` to seccomp for all syscalls that are handled by Landlock, so we don't go through both a Landlock check _and_ a ptrace check.

We can detect the presence of Landlock at runtime by issuing `sys_landlock_create_ruleset` and checking that the errno is not `ENOSYS`.

Reply:
Alright, sounds good

Reply:
I was thinking of having a list of system call numbers prepared for landlock, and allowing those if landlock creation succeeds. This would likely result in additional plumbing, but has the advantage of doing detection at the exact spot where it's used.

Reply:
The downside of that is it's a little surprising to have handlers listed in `isolate.py` that never end up getting called. I don't think specifying a handler that's also handled by landlock should be disallowed (e.g., might want to do so just to log all filesystem accesses), it's just not particularly useful.

Reply:
We'd want to mark handlers as conditional on landlock or something then, and export that down to the landlock setup code